src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -289,6 +291,9 @@ public class KNNSettings {
         }
     };
 
+    private final static Map<String, Setting<?>> FEATURE_FLAGS = getFeatureFlags().stream()
+        .collect(toUnmodifiableMap(Setting::getKey, Function.identity()));
```

Comment:
do we need this? can we just use the functions directly on line 402 of this file?

Reply:
Yes we need this, this is used in line 354 when we are trying to get the settings